### PR TITLE
[Protocol3] Add basic support for offchain data-availability

### PR DESCRIPTION
**Pull request by dong77**

### CODE REVIEW

File: packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol [hunk 1]
```diff
@@ -54,10 +54,15 @@ library ExchangeBlocks
...
         uint16 blockSize,
-        bytes memory data
+        bytes memory data,
+        bytes memory offchainData
         )
         internal  // inline call
     {
+        if (S.onchainDataAvailability) {
+            require(offchainData.length == 0, "INVALID_OFFCHAIN_DATA");
+        }
```
Comment (dong77) [inline packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol]: when onchainDataAvailabile == false, we do not enforce offchainData.length > 0

File: packages/loopring_v3/contracts/test/Operator.sol [hunk 2]
```diff
@@ -38,7 +38,7 @@ contract Operator {
         )
         external
     {
-        exchange.commitBlock(blockType, numElements, data);
+        exchange.commitBlock(blockType, numElements, data, bytes("0"));
```
Comment (dong77) [inline packages/loopring_v3/contracts/test/Operator.sol]: not sure if bytes("0").length == 0.

**Comment by Brechtpd** [comment]
Related to #139. Will need some minor modifications to the circuits.

**Comment by dong77** [comment]
@Brechtpd I’m merging this PR and leave the modification of circuits to you.

**Pull request merged**